languages/erlang/exercises/concept/basics/.docs/introduction.md

```diff
@@ -0,0 +1,60 @@
+To define a variable and bind a value to it, use the `=` operator:
+
+```erlang
+MyValue = 10.
+```
+
+Variable names must begin with an uppercase letter and are generally written in camel case.
+
+To enforce [immutabilty][immutability], once a variable is bound to a value, it cannot be updated:
+
+```erlang
+Count = 5,
+Count = Count + 1.
+% exception error: no match of right hand side value 6
+```
+
+The `=` operator is also used for performing [pattern matching][pattern_matching] to bind values to variables. In the case above, `Count` is bound to the value `5`, then pattern matching is performed with `6` on the right side being matched to `5` on the left. `5` is not matched with `6`, so an error occurs. [Pattern matching][pattern_matching] will be discussd in more detail in later exercises.
```

Comment:
```suggestion
The `=` operator is also used for performing [pattern matching][pattern_matching] to bind values to variables. In the case above, `Count` is bound to the value `5`, then pattern matching is performed with `6` on the right side being matched to `5` on the left. `5` is not matched with `6`, so an error occurs. [Pattern matching][pattern_matching] will be discussed in more detail in later exercises.
```

Reply:
In trying to keep the introduction.md as minimal as possible, I'm wondering if this section could maybe removed here and have it only in the after.md document? Note that the introduction document's goal is to provide information that gives the student just enough context to figure out the solution themselves. Anything that is not essential for the student to be able to solve the solution should be omitted from the introduction.md (it can be included in the after.md document though).

Reply:
I'm also in favor of droping this paragraph and leaving the error as is, just showing that rebinding isn't possible.

For pattern matching there should be a distinct concept exercise that then explains what happens in the example above.